### PR TITLE
Automate Ensemble Setup with ensembles_scaffold_builder.py for Directory and Script Management

### LINKED ISSUE #105: Automate Ensemble Setup with `ensembles_scaffold_builder.py` for Directory and Script Management
### Issue: Automate Ensemble Setup with `ensembles_scaffold_builder.py` for Directory and Script Management

**Description**  
Develop a Python script, `ensembles_scaffold_builder.py`, to automate the creation and management of directory structures and configuration scripts for ensemble models. This script will enforce consistent setup standards, paralleling the model scaffold builder while meeting unique ensemble requirements.

**Objectives and Requirements**

1. **Initialization for Ensemble**:
   - Adapt the `ModelScaffoldBuilder` structure to `EnsembleScaffoldBuilder`, initializing with `ensemble_name` instead of `model_name`.
   - Implement ensemble-specific validation, ensuring names are unique and meet naming conventions (e.g., lowercase `adjective_noun`).
   - Implement namespace separation to avoid conflicts between ensemble and model names.

2. **Directory and Script Creation**:
   - Create ensemble directories in a similar hierarchy to the model directories within `views_pipeline/ensembles/[ensemble_name]`.
   - Initialize subdirectories and scripts specific to ensemble requirements
   - 
3. **Assessment Functions**:
   - Implement something like `assess_ensemble_directory()` and `assess_ensemble_scripts()` to validate directory and script presence, logging any missing elements to ensure a complete setup.
   - Ensure assessment results provide detailed logs for missing directories and scripts, triggering warnings or errors as appropriate.

4. **Logging and Error Handling**:
   - Integrate Python’s `logging` module for consistent logging practices across the pipeline.
   - Ensure that `ERROR` levels halt the script and provide clear messages, especially if required directories or scripts are missing.

5. **Interactive Naming Validation**:
   - Incorporate a way to validate and confirm ensemble names, cross-checking against both existing model and ensemble names for uniqueness.
   - Ensure names follow lowercase `adjective_noun` format to meet pipeline standards.

**Tasks**
- [x] Define the `EnsembleScaffoldBuilder` class, mirroring structure from `ModelScaffoldBuilder` but tailored for ensemble needs.
- [x] Implement directory and script generation functions with ensemble-specific configurations (e.g., `config_ensemble_meta`, `config_weights`).
- [x] Develop assessment methods to check the presence of ensemble directories and scripts, logging any missing elements.
- [x] Add validation for unique and properly formatted ensemble names.
- [ ] Document the `ensembles_scaffold_builder.py` script, providing usage instructions, examples, and error-handling guidelines.
- [ ] Test the script end-to-end, including edge cases, to verify it meets directory creation, script generation, and logging requirements.

**Next Steps**  
Once `ensembles_scaffold_builder.py` is complete and tested, create a usage guide and integrate it with the pipeline documentation. 

**Labels**  
`script`, `ensemble`, `MLOps`, `directory management`, `logging`

### PR DESCRIPTION
**Pull request by smellycloud**

### CODE REVIEW

File: common_utils/ensemble_path.py [hunk 1]
```diff
@@ -36,11 +36,115 @@ def __init__(
...
+        """
+        self.scripts = [
+            self._build_absolute_directory(Path("configs/config_deployment.py")),
+            self._build_absolute_directory(Path("configs/config_hyperparameters.py")),
+            self._build_absolute_directory(Path("configs/config_meta.py")),
+            self._build_absolute_directory(Path("main.py")),
+            self._build_absolute_directory(Path("README.md")),
+            self._build_absolute_directory(Path("requirements.txt")),
+            self._build_absolute_directory(Path("artifacts/model_metadata_dict.py")),
+            self._build_absolute_directory(Path("src/dataloaders/get_data.py")),
```
Comment (xiaolong0728) [inline common_utils/ensemble_path.py:126]: Ensemble doesn't require get_data

File: common_utils/ensemble_path.py [hunk 2]
```diff
@@ -36,11 +36,115 @@ def __init__(
...
+            self._build_absolute_directory(
+                Path("src/management/execute_model_tasks.py")
+            ),
+            self._build_absolute_directory(
+                Path("src/offline_evaluation/evaluate_ensemble.py")
+            ),
+            self._build_absolute_directory(Path("src/training/train_ensemble.py")),
+            self._build_absolute_directory(Path("src/utils/utils_outputs.py")),
+            self._build_absolute_directory(Path("src/utils/utils_run.py")),
+            self._build_absolute_directory(Path("src/visualization/visual.py")),
```
Comment (xiaolong0728) [inline common_utils/ensemble_path.py:142]: utils_check.py in src/utils

**Review by xiaolong0728 — COMMENTED** [review]
It looks good to me. Only some small changes are required.

**Review by Polichinel — APPROVED** [review]
looks fine
